File: numberGuess.c
#include<math.h>
#include<stdio.h>
#include<stdlib.h>
#include<time.h>

int main()
{
    int start, end, x, guess, count = 0, flag = 0, tot_chan;
    printf("Enter start number: ");
    scanf("%d", &start);
    printf("Enter end number: ");
    scanf("%d", &end);
    // srand(time(NULL));
    x = (rand() % (end - start + 1)) + start;
    tot_chan = log(end - start + 1) / log(2);

    printf("\n\tYou've only %d chances to guess the integer!\n\n", tot_chan);
    while (count < tot_chan) {
        count++;
        printf("Guess a number: ");
        scanf("%d", &guess);
        if (x == guess) {
            printf(
                "Congratulations you did it in %d try!\n",
                count);
            flag = 1;
            break;
        }
        else if (x > guess) {
            printf("You guessed too small!\n");
        }
        else if (x < guess) {
            printf("You guessed too high!\n");
        }
    }
    if (!flag) {
        printf("\nThe number is %d\n", x);
        printf("\tBetter Luck Next time!\n");
    }

    return 0;
}

File: numberGuess.py
import math
import random

#Taking inputs
start = int(input("Enter starting number : "))
end = int(input("Enter ending number : "))

#Generating random number
n = random.randint(start,end)

#Calculating chances
chance = int(math.log(end-start+1)/math.log(2))

print('Total number of guesses : ', chance , '\n')

#Guessing number
count = 0
flag = 0
while(count < chance):
    count += 1
    guess=int(input('Guess the number : '))
    if(guess == n):
        print('You guessed the number correct on ' , count , '\n')
        flag = 1
        break
    elif(guess < n):
        print('You guess too small') 
    else:
        print('You guessed too high')
if(flag != 1):
    print('The number is ' , n , '\nBetter luck next time\n')
